frontend/src/queries/Query/Query.tsx

```diff
@@ -74,7 +74,7 @@ export function Query<Q extends Node>(props: QueryProps<Q>): JSX.Element | null
     }, [propsQuery])
 
     const query = readOnly ? propsQuery : localQuery
-    const setQuery = readOnly ? undefined : propsSetQuery ?? localSetQuery
+    const setQuery = propsSetQuery ?? localSetQuery
```

Comment:
Might have missed something, why do we not need to check for `readOnly` anymore?

Reply:
Because I _always_ want `setQuery` set from props if props are set - therefore we no longer need the ternary

Reply:
Ok, I would
1. Make sure to test that when viewing insights and other things in `readOnly` mode and then toggling the query (like date range) it doesn't make those persist. For components like `DataTable` it figures out if it's in `readOnly` mode using `setQuery` like `const isReadOnly = setQuery === undefined`

2. Update types for `InsightViz, DataTable...` and other comps to ensure `setQuery` can't be `undefined`